RFC-0004-sparse-fill-value/README.md

```diff
@@ -0,0 +1,509 @@
+# PyTorch Sparse Tensors: fill-value property
+
+|            |                 |
+| ---------- | --------------- |
+| Authors    | Pearu Peterson  |
+| Status     | Draft           |
+| Type       | Process         |
+| Created    | 2020-09-08      |
+| Resolution | TBD             |
+
+## Abstract
+
+This proposal introduces a fill-value property to PyTorch sparse
+tensors that generalizes the current interpretation of unspecified
+elements from zero value to any value, including undefined value as a
+future extension.
+
+## Motivation and Scope
+
+In general, the unspecified elements of sparse tensors have
+domain-specific interpretations:
+- In Linear Algebra domain, the unspecified elements are zero valued.
+- In Graph domain, the unspecified elements of adjacency matrices
+  represent non-edges. In this document, a non-edge corresponds to
+  undefined value.
+- In neural networks, sparse tensors can be inputs, say, to activation
+  functions that are defined in terms of elementary functions.
+  Evaluation of such functions on sparse tensors can be element-wise,
+  or one-dimension-wise operations. This includes the evaluation of
+  the functions on the values corresponding to unspecified
+  elements. Even when the unspecified elements are initially being
+  zero valued, the values of unspecified elements may be mapped to
+  non-zero values. So, in the domain of Calculus, the unspecified
+  elements of sparse tensors can have any defined value.
```

Comment:
The neuroscience use case towards the end of this RFC is a good motivating use case. It would be useful to link to it from this section.

Reply:
Good idea. Done.